azure/converters/image.go

```diff
@@ -49,16 +49,38 @@ func mpImageToSDK(image *infrav1.Image) (*compute.ImageReference, error) {
 	}, nil
 }
 
-func sigImageToSDK(image *infrav1.Image) (*compute.ImageReference, error) {
-	imageID := fmt.Sprintf("/subscriptions/%s/resourceGroups/%s/providers/Microsoft.Compute/galleries/%s/images/%s/versions/%s",
-		image.SharedGallery.SubscriptionID,
-		image.SharedGallery.ResourceGroup,
-		image.SharedGallery.Gallery,
-		image.SharedGallery.Name,
-		image.SharedGallery.Version)
+func computeImageToSDK(image *infrav1.Image) (*compute.ImageReference, error) {
+	idTemplate := "/subscriptions/%s/resourceGroups/%s/providers/Microsoft.Compute/galleries/%s/images/%s/versions/%s"
+
+	if image.SharedGallery != nil {
+		return &compute.ImageReference{
+			ID: to.StringPtr(fmt.Sprintf(idTemplate,
+				image.SharedGallery.SubscriptionID,
+				image.SharedGallery.ResourceGroup,
+				image.SharedGallery.Gallery,
+				image.SharedGallery.Name,
+				image.SharedGallery.Version,
+			)),
+		}, nil
+	}
+
+	if image.ComputeGallery.ResourceGroup != nil {
```

Comment:
If a user is trying to specify a private image and they leave out the resource group, what happens? Does it fail to provision the VM because it can't find the image?

I am thinking a log here might be useful to someone debugging a scenario like that

Reply:
>If a user is trying to specify a private image and they leave out the resource group, what happens?

If subscription ID is specified or other way around (resource group given but no subscription ID), then validating webhook returns an error. If neither is specified, Gallery name is assumed to be the community gallery public name.

Do you think community gallery public name should have it's own field perhaps to make it more explicit? But then the only shared fields becomes plan and version, so we could as well introduce another type for the community galleries (which on it's own is not such a bad idea I guess and SIG -> Compute Gallery cleanup is still nice to have. 

@CecileRobertMichon since you suggested to extract common fields in this PR (as original version was adding only separate community gallery field), what do you think?

Reply:
Hmm, I also had a look into https://github.com/hashicorp/packer-plugin-azure/pull/199, which is also about adding support for the directly shared galleries, which use yet another ID field. I think we should at least make the new type forward-compatible with those as well. As far as I understand, directly shared galleries also does not require subscription ID or resource group, which breaks my assumptions taken here.

Seems like directly shared galleries are referred using `SharedGalleryImageId` field in `ImageReference` field.

I guess we could introduce something like `Type` field let user specify how the image is shared or something.

Reply:
I was thinking, if we could stat if a given Directly shared image exists or if a Community shared image is accessible, then we could save a hassle for the users to specify that and perhaps only require `sharingMethod` (new name which perhaps make most sense? I'm still trying to figure it out!) field when the information is ambiguous. 

Code like this seem to work just fine cross-tenants:
```go
  communityGalleryImage, err = client.Get(ctx, "westeurope", badPublicGalleryName, galleryImageName, imageVersion)
  if err != nil {
    derr := autorest.DetailedError{}
    if errors.As(err, &derr) && derr.StatusCode != 404 {
      panic(err)
    }

    fmt.Println("Image correctly not found")
  }
```